### PR TITLE
Kernel: Use more fine-grained content data block granularity in TmpFS 

### PR DESCRIPTION
**Pull request by supercomputer7**
Relies on #14785.

### CODE REVIEW

File: Kernel/FileSystem/Ext2FileSystem.cpp [hunk 1]
```diff
@@ -817,8 +825,12 @@ ErrorOr<size_t> Ext2FSInode::read_bytes(off_t offset, size_t count, UserOrKernel
         return nread;
     }
 
-    if (m_block_list.is_empty())
-        m_block_list = TRY(compute_block_list());
+    // Note: We bypass the const declaration of this method, but this is a strong
+    // requirement to be able to accomplish the read operation successfully.
+    // We call this special method becuase it locks a separate mutex to ensure we
```
Comment (Hendiadyoin1) [inline Kernel/FileSystem/Ext2FileSystem.cpp:830]: > becuase

Comment (supercomputer7) [reply Kernel/FileSystem/Ext2FileSystem.cpp]: Fixed in #15370.

**Comment by supercomputer7** [comment]
Fixes #14560.

**Comment by supercomputer7** [comment]
This essentially opens the way to utilize huge files on x86-64 port, because right now the kmalloc code asserts when a file (or a bunch of them) reach 3.9 GiB with `Out of address space when expanding kmalloc heap.` error and with a kernel panic.
This should be fixed later on, but I am glad that there's finally an opportunity to fix x86-64 issues on allocation of pages beyond the 4 GiB barrier :)

**Comment by supercomputer7** [comment]
This failed about 3 times already due to CI flakes, so I am force-pushing one last time in hope it will pass CI now.

**Comment by supercomputer7** [comment]
I forgot to mention, but this is vital for #14936, because if we put an entire filesystem on the initramfs image, we want to be sure that virtual memory space is not exhausted.

**Pull request merged**